### PR TITLE
add scanner and lexer for DSL

### PR DESCRIPTION
**Pull request by oliverselinger**

### CODE REVIEW

File: src/main/java/org/nobloat/bare/dsl/Scanner.java [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+            }
+
+            builder.append(character);
+        }
+
+        return builder.toString();
+    }
+
+    public boolean isSpace(char character) {
+        return Character.isSpaceChar(character);
```
Comment (cinemast) [inline src/main/java/org/nobloat/bare/dsl/Scanner.java:99]: I think we can call these methods directly.

File: src/main/java/org/nobloat/bare/dsl/Lexer.java [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+            this(type, null);
+        }
+
+        public Token(Type type, String value) {
+            this.type = type;
+            this.value = value;
+        }
+
+        enum Type {
+            INTEGER,
```
Comment (cinemast) [inline src/main/java/org/nobloat/bare/dsl/Lexer.java:132]: may warp them into fewer lines?

File: src/main/java/org/nobloat/bare/dsl/Lexer.java [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+            EQUAL,
+            COLON,
+            UNKNOWN,
+            EOF,
+            COMMENT,
+
+            TYPE,
+            ENUM,
+            UINT,
+            U8,
```
Comment (cinemast) [inline src/main/java/org/nobloat/bare/dsl/Lexer.java:147]: I think the number types could at least be grouped together.

**Review by cinemast — APPROVED** [review]

File: src/main/java/org/nobloat/bare/dsl/Scanner.java (hunk 1 above)
Comment (oliverselinger) [reply src/main/java/org/nobloat/bare/dsl/Scanner.java]: agree, done!

File: src/main/java/org/nobloat/bare/dsl/Lexer.java (hunk 3 above)
Comment (oliverselinger) [reply src/main/java/org/nobloat/bare/dsl/Lexer.java]: agree, done!

**Pull request merged**